SharpRepository.Repository/Caching/StandardCachingStrategyBase.cs

```diff
@@ -428,7 +428,12 @@ private int GetGeneration()
         {
             if (!GenerationalCachingEnabled) return 1; // no need to use the caching provider
 
-            return !CachingProvider.Get(GetGenerationKey(), out int generation) ? 1 : generation;
+            if (!CachingProvider.Get(GetGenerationKey(), out int generation))
+            {
+                IncrementGeneration();
+                return GetGeneration();
```

Comment:
couldn't this be simplified to `return IncrementGeneration();` since it returns what that key is there shouldn't be a need to ask for it again.